test/OperatorTest.jl

```diff
@@ -180,3 +180,11 @@ A=[dirichlet(S);Derivative(S)^2 - exp(im*x)]
 QR=qrfact(A)
 
 @test norm((A\[1.])-(QR\[1.]))<100eps()
+
+# 2d derivative (issue #346)
+let d = Space([0,1]) * Space([0,2]),
+    D = Derivative(d, [0,1]),
+    f = Fun((x,y) -> sin(x) * cos(y), d),
+    fd = Fun((x,y) -> -sin(x) * sin(y), d)
+    @test (D*f)(0.2,0.3) ≈ fd(0.2,0.3)
```

Comment:
Note that what I really wanted to do was `@test norm(D*f - fd) < 100*eps()`, but the expression `D*f - fd` gives
```
ERROR: AssertionError: iskronsumop(A)
 in schurfact at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/factorizations.jl:197
 in schurfact at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/factorizations.jl:215
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:177
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:135
 in pdesolve at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:179
 in \ at /Users/stevenj/.julia/v0.4/ApproxFun/src/PDE/pdesolve.jl:187
 in defaultcoefficients at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Space.jl:351
 in coefficients at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Fun.jl:46
 in call at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/constructors.jl:66
 in - at /Users/stevenj/.julia/v0.4/ApproxFun/src/Fun/Fun.jl:197
```
Aren't we supposed to be able to subtract functions and derivatives?